### PR TITLE
feature (auth/ldap): service proto definition

### PR DESCRIPTION
**Pull request by jimlambrt**
Define service protobufs for AuthMethod, Account and ManagedGroups

### CODE REVIEW

File: internal/proto/controller/api/resources/authmethods/v1/auth_method.proto [hunk 1]
```diff
@@ -300,3 +305,255 @@ message OidcAuthMethodAuthenticateTokenResponse {
...
+  // discover_dn if true, use anon bind to discover the bind DN of a user.
+  // Defaults to false.
+  bool discover_dn = 40 [
+    json_name = "discover_dn",
+    (custom_options.v1.generate_sdk_option) = true,
+    (custom_options.v1.mask_mapping) = {
+      this: "attributes.discover_dn"
+      that: "DiscoverDn"
+    }
+  ];
```
Comment (irenarindos) [inline internal/proto/controller/api/resources/authmethods/v1/auth_method.proto:346]: Missing gotags?

File: internal/proto/controller/api/resources/authmethods/v1/auth_method.proto [hunk 2]
```diff
@@ -300,3 +305,255 @@ message OidcAuthMethodAuthenticateTokenResponse {
...
+  // anon_group_search if true, use anon bind when performing LDAP group
+  // searches. Defaults to false.
+  bool anon_group_search = 50 [
+    json_name = "anon_group_search",
+    (custom_options.v1.generate_sdk_option) = true,
+    (custom_options.v1.mask_mapping) = {
+      this: "attributes.anon_group_search"
+      that: "AnonGroupSearch"
+    }
+  ];
```
Comment (irenarindos) [inline internal/proto/controller/api/resources/authmethods/v1/auth_method.proto:357]: Missing gotags?

File: internal/proto/controller/api/resources/authmethods/v1/auth_method.proto (hunk 1 above)
Comment (jimlambrt) [reply internal/proto/controller/api/resources/authmethods/v1/auth_method.proto:346]: ty!

**Review by irenarindos — APPROVED** [review]
Looks good!

I'm unsure if this is something we need to be concerned about though? "Unchanged files with check annotations": https://github.com/hashicorp/boundary/pull/2761/files#diff-bcf83b5283886c98fc45b6435fdc94d87d545b2768c6beac8acd14eb9d6b13f1

File: internal/proto/controller/api/resources/authmethods/v1/auth_method.proto [hunk 3]
```diff
@@ -300,3 +305,255 @@ message OidcAuthMethodAuthenticateTokenResponse {
...
+    (custom_options.v1.mask_mapping) = {
+      this: "attributes.group_filter",
+      that: "GroupFilter"
+    }
+  ]; // @gotags: `class:"public"`
+
+  // certificates are optional PEM encoded x509 certificates in ASN.1 DER form
+  // that can be used as trust anchors when connecting to an LDAP provider.
+  // These are Value Objects that will be stored as Certificate messages, and
+  // are operatated on as a complete set (not individually).
```
Comment (johanbrandhorst) [inline internal/proto/controller/api/resources/authmethods/v1/auth_method.proto:474]:
```suggestion
  // are operated on as a complete set (not individually).
```

**Review by johanbrandhorst — APPROVED** [review]

File: internal/proto/controller/api/resources/authmethods/v1/auth_method.proto (hunk 3 above)
Comment (jimlambrt) [reply internal/proto/controller/api/resources/authmethods/v1/auth_method.proto:474]: ty!

**Pull request merged**